context/AuthContext.tsx

```diff
@@ -0,0 +1,41 @@
+import {createContext, ReactNode, useContext, useEffect, useState} from 'react';
+import { getAuth, onAuthStateChanged, User } from 'firebase/auth';
+import firebase_app from '@/firebase/config';
+
+const auth = getAuth(firebase_app);
+
+interface AuthContextType {
+    user: User | null;
+}
+
+export const AuthContext = createContext<AuthContextType>({
+    user: null,
+});
+
+export const useAuthContext = () => useContext(AuthContext);
+
+interface AuthContextProviderProps {
+    children: ReactNode;
+}
+
+export const AuthContextProvider = ({ children }: AuthContextProviderProps) => {
+    const [user, setUser] = useState<User | null>(null);
+    const [loading, setLoading] = useState<boolean>(true);
+
+    useEffect(() => {
+        const unsubscribe = onAuthStateChanged(auth, (user) => {
+            setUser(user);
+            setLoading(false);
+        });
+        return () => unsubscribe();
+    }, []);
+
+
+    return (
+        <AuthContext.Provider value={{ user }}>
```

Comment:
ik zou misschien gewoon hier die sign in en sign out functions definiëren en ze mee geven aan de provider. 
dan kan je ze vervolgens aan roepen door const {signIn}=useAuthContext() te definieren. dan kan je gewoon sign in aan roepen.